File: insertion_sort/insertion_sort.rb
def insertion_sort(input)
  for i in 1..input.length-1
    key = input[i]
    j = i - 1
    while j >= 0 and input[j] > key
      input[j+1] = input[j]
      j = j - 1
    end
    input[j+1] = key
  end
  return input
end



rand = Random.new
input = Array.new(500) { |num| rand.rand(500) }

p insertion_sort(input)

File: insertion_sort/insertion_sort.py
import random


def insertion_sort(inp):
	for i in range(1,len(inp)):
		key = inp[i]
		j = i - 1
		while j >= 0 and inp[j] > key:
			inp[j+1] = inp[j]
			j = j - 1
		inp[j+1] = key
	return inp

if __name__ == '__main__':
    inp = random.sample(range(100), 50)
    print(inp)
    insertion_sort(inp)
    print(inp)
